src/subcommands/subcommand.go

```diff
@@ -186,20 +218,31 @@ func executeAdd(args []string) {
 		*sourceType = sourceTypeStr
 	}
 
-	addArgs := subcommands.AddArgs{
+	dataUris, err := convertUrisToAbsPath(cmd.Args(), *sourceType == "local_file")
+	if err != nil {
+		printCmdErr(err.Error())
+	}
+
+	addArgs := &pb.AddRequest{
 		IndexPath:  *indexPath,
 		DataType:   *dataType,
 		SourceType: *sourceType,
 		MakeCopy:   *makeLocalCopy,
-		DataURIs:   cmd.Args(),
+		DataUris:   dataUris,
 	}
 
-	subcommands.Add(addArgs)
+	res, err := grpcclient.Add(addArgs)
```

Comment:
Refactor Add in subcommands/add.go to take AddRequest (the proto, not the struct) as an argument and instead simply call Add without having created the grpcclient.

i.e.

Add(a MyAddProto)

If the actual grpc method is called Add you may wish to rename Add in subcommand.go to AddInternal or something.

Basically we should leave subcommand.go almost exactly the same as it is now, and just pass Protobufs to the subcommands instead of the (very similar/same) go structs we use currently on main.